File: web/console-fe/src/services/UserService.ts
import {HttpCommon} from "@/common/HttpCommon";
import {ResponseModel} from "@/model/ResponseModel";
import {UserPassword} from "@/model/UserPassword";
import {UserName} from "@/model/UserName";
import {Filter} from "@/model/Filter";

const baseUrl = "/api/v1/user";

class UserService
{
  getInfo(): Promise<ResponseModel>
  {
    return new HttpCommon().get(baseUrl);
  }

  changePassword(configure: UserPassword): Promise<ResponseModel>
  {
    return new HttpCommon().put(baseUrl + '/changePassword', configure);
  }

  changeUsername(configure: UserName): Promise<ResponseModel>
  {
    return new HttpCommon().put(baseUrl + '/changeUsername', configure);
  }

  getLogs(filter: Filter): Promise<ResponseModel>
  {
    return new HttpCommon().post(baseUrl + '/log', filter);
  }
}

export default new UserService();

File: core/datacap-service/src/main/java/io/edurt/datacap/service/service/UserService.java
package io.edurt.datacap.service.service;

import io.edurt.datacap.common.response.CommonResponse;
import io.edurt.datacap.service.body.FilterBody;
import io.edurt.datacap.service.body.UploadBody;
import io.edurt.datacap.service.body.UserNameBody;
import io.edurt.datacap.service.body.UserPasswordBody;
import io.edurt.datacap.service.entity.PageEntity;
import io.edurt.datacap.service.entity.UserEntity;
import io.edurt.datacap.service.entity.convert.AvatarEntity;
import io.edurt.datacap.service.entity.itransient.user.UserEditorEntity;
import io.edurt.datacap.service.model.AiModel;
import io.edurt.datacap.service.record.TreeRecord;

import java.util.List;

public interface UserService
        extends BaseService<UserEntity>
{
    CommonResponse<UserEntity> saveOrUpdate(UserEntity configure);

    CommonResponse<Object> authenticate(UserEntity configure);

    CommonResponse<UserEntity> info(String code);

    CommonResponse<Long> changePassword(UserPasswordBody configure);

    CommonResponse<Long> changeUsername(UserNameBody configure);

    /**
     * Check whether the given username can be used for renaming:
     * format valid (4-20 chars, letters/numbers/underscores, not starting
     * with a digit) and not taken by another account.
     */
    CommonResponse<Boolean> checkUsername(String username);

    CommonResponse<Long> changeThirdConfigure(AiModel configure);

    CommonResponse<List<Object>> getSugs(Long id);

    CommonResponse<List<TreeRecord>> getMenus();

    CommonResponse<PageEntity<UserEntity>> getAll(FilterBody filter);

    CommonResponse<UserEntity> changeEditorConfigure(UserEditorEntity configure);

    CommonResponse<AvatarEntity> uploadAvatar(UploadBody configure);

    CommonResponse<UserEntity> changeNotify(UserEntity configure);
}
